sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/Disk.java

```diff
@@ -153,6 +155,48 @@ public interface Disk extends GroupableResource<ComputeManager, DiskInner>, Refr
      */
     PublicNetworkAccess publicNetworkAccess();
 
+    /**
+     * Gets the number of IOPS allowed for this disk.
+     *
+     * @return the number of IOPS allowed for this disk.
+     */
+    Long diskIopsReadWrite();
+
+    /**
+     * Gets the throughput (MBps) allowed for this disk.
+     *
+     * @return the throughput (MBps) allowed for this disk.
+     */
+    Long diskMBpsReadWrite();
+
+    /**
+     * Gets the total number of IOPS allowed across all VMs mounting this shared disk as read-only.
+     *
+     * @return the total number of IOPS allowed for this shared read-only disk.
+     */
+    Long diskIopsReadOnly();
+
+    /**
+     * Gets the total throughput (MBps) allowed across all VMs mounting this shared disk as read-only.
+     *
+     * @return the total throughput (MBps) allowed for this shared read-only disk.
+     */
+    Long diskMBpsReadOnly();
+
+    /**
+     * Gets the maximum number of VMs that can attach to the disk at the same time.
+     *
+     * @return the maximum number of VMs that can attach to the disk at the same time.
+     */
+    int maximumShares();
+
+    /**
+     * Gets the information about the VMs that have the disk attached.
+     *
+     * @return the information about the VMs that have the disk attached.
+     */
+    List<ShareInfoElement> shareInfo();
```

Comment:
nit, maybe `currentShareInfo`, or `currentShares`?

Reply:
However, it seems always `null`. Still checking why.

Reply:
Guess I will just remove it.

Portal seems got it via
```
  "managedBy": "/subscriptions/faa080af-c1d8-40ad-9cce-e1a450ca5b57/resourceGroups/javacsmrg02664/providers/Microsoft.Compute/virtualMachines/javavm2",
  "managedByExtended": [
    "/subscriptions/faa080af-c1d8-40ad-9cce-e1a450ca5b57/resourceGroups/javacsmrg02664/providers/Microsoft.Compute/virtualMachines/javavm2",
    "/subscriptions/faa080af-c1d8-40ad-9cce-e1a450ca5b57/resourceGroups/javacsmrg02664/providers/Microsoft.Compute/virtualMachines/javavm"
  ],
```